readme.md

```diff
@@ -0,0 +1,278 @@
+
+const apiLimiter = slowDown({
+	windowMs: 15 * 60 * 1000, // 15 minutes
+	delayAfter: 1, // Allow only one request to go at full-speed.
+	delayMs: (hits) => hits * hits * 1000, // 2nd request has a 4 second delay, 3rd is 9 seconds, 4th is 16, etc.
+})
+
+// Apply the delay middleware to API calls only.
+app.use('/api', apiLimiter)
+```
+
+To use a custom store:
+
+```ts
+import { slowDown } from 'express-slow-down'
+import { MemcachedStore } from 'rate-limit-memcached'
+
+const speedLimiter = slowDown({
+	windowMs: 15 * 60 * 1000, // 15 minutes
+	delayAfter: 1, // Allow only one request to go at full-speed.
+	delayMs: (hits) => hits * hits * 1000, // Add exponential delay after 1 request.
+	store: new MemcachedStore({
+		/* ... */
+	}), // Use the external store
+})
+
+// Apply the rate limiting middleware to all requests.
+app.use(speedLimiter)
+```
+
+> **Note:** most stores will require additional configuration, such as custom
+> prefixes, when using multiple instances. The default built-in memory store is
+> an exception to this rule.
+
+## Configuration
+
+### [`windowMs`](https://github.com/express-rate-limit/express-rate-limit/blob/main/readme.md#windowms)
+
+> `number`
+
+Time frame for which requests are checked/remembered.
+
+Note that some stores have to be passed the value manually, while others infer
+it from the options passed to this middleware.
+
+Defaults to `60000` ms (= 1 minute).
+
+### `delayAfter`
+
+> `number` | `function`
+
+The max number of requests allowed during `windowMs` before the middleware
+starts delaying responses. Can be the limit itself as a number or a (sync/async)
+function that accepts the Express `req` and `res` objects and then returns a
+number.
+
+Defaults to `1`.
+
+An example of using a function:
+
+```ts
+const isPremium = async (user) => {
+	// ...
+}
+
+const limiter = slowDown({
+	// ...
+	delayAfter: async (req, res) => {
+		if (await isPremium(req.user)) return 10
+		else return 1
+	},
+})
+```
+
+### `delayMs`
+
+> `number | function`
+
+The delay to apply to each request once the limit is reached. Can be the delay
+itself (in milliseconds) as a number or a (sync/async) function that accepts a
+number (number of requests in the current window), the Express `req` and `res`
+objects and then returns a number.
+
+By default, it increases the delay by 1 second for every request over the limit:
+
+```ts
+const limiter = slowDown({
+	// ...
+	delayMs: (used) => (used - delayAfter) * 1000,
+})
+```
+
+### `maxDelayMs`
+
+> `number | function`
+
+The absolute maximum value for `delayMs`. After many consecutive attempts, the
+delay will always be this value. This option should be used especially when your
+application is running behind a load balancer or reverse proxy that has a
+request timeout.
+
+Defaults to `Infinity`.
+
+For example, for the following configuration:
+
+```ts
+const limiter = slowDown({
+	// ...
+	delayAfter: 1,
+	delayMs: (hits) => hits * 1000,
+	maxDelayMs: 4_000,
```

Comment:
Not sure I like the underscore here... did you do that, or was it an eslint rule?

```suggestion
	maxDelayMs: 4000,
```